app/models/comment.rb

```diff
@@ -1,2 +1,15 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_create :update_post_comments_counter
+  before_destroy :update_comments_before_destroy
+
+  def update_post_comments_counter
+    post.increment!(:commentscounter)
+  end
+
+  def update_comments_before_destroy
+    post.decrement!(:commentscounter)
+  end
```

Comment:
- [ ] In your `Comment` model, consider making the `update_post_comments_counter` and `update_comments_before_destroy` methods private. By marking them as private, you signify that these methods are meant for internal use within the `Comment` model and should not be accessed or called from outside the class. This encapsulation helps maintain clean and organized code, preventing unintended usage and potential issues in the future.

```suggestion
private

  def update_post_comments_counter
    post.increment!(:commentscounter)
  end

  def update_comments_before_destroy
    post.decrement!(:commentscounter)
  end
```